easybuild/easyblocks/e/esmf.py

```diff
@@ -136,22 +136,36 @@ def install_step(self):
 
         python = get_software_version('Python')
         if python:
-            # then, install the python bindings
-            py_subdir = os.path.join(self.builddir, 'esmf-ESMF_%s' % '_'.join(self.version.split('.')),
-                                     'src', 'addon', 'ESMPy')
+            if LooseVersion(self.version) < LooseVersion('8.4.0'):
+                # then, install the python bindings
+                py_subdir = os.path.join(self.builddir, 'esmf-ESMF_%s' % '_'.join(self.version.split('.')),
+                                         'src', 'addon', 'ESMPy')
+            else:
+                # then, install the python bindings
+                py_subdir = os.path.join(self.builddir, 'esmf-%s' % self.version,
+                                         'src', 'addon', 'esmpy')
             try:
                 os.chdir(py_subdir)
             except OSError as err:
                 raise EasyBuildError("Failed to move to: %s", err)
 
-            cmd = "python setup.py build --ESMFMKFILE=%s/lib/esmf.mk " % self.installdir
-            cmd += " && python setup.py install --prefix=%s" % self.installdir
-            run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+            if LooseVersion(self.version) < LooseVersion('8.4.0'):
+                cmd = "python setup.py build --ESMFMKFILE=%s/lib/esmf.mk " % self.installdir
+                cmd += " && python setup.py install --prefix=%s" % self.installdir
+                run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+            else:
+                cmd = "ESMFMKFILE=%s/lib/esmf.mk " % self.installdir
+                cmd += "python -m pip install . --no-deps --prefix=%s" % self.installdir
```

Comment:
EB also adds pip options `--ignore-installed  --no-index  --no-build-isolation`, shouldn't those be added here too?

Reply:
not tested

Reply:
I agree with @smoors that we should add those additional options for `pip install`, they're there for a reason (and it's better to err on the side of caution with installing Python packages).

It would be nicer if we could query `PythonPackage` for which options to pass to `pip install` though...